### DISCUSSION TITLE
Access Axios global variable

### DISCUSSION THREAD

**Discussion by fgb4101** [alpine-collective/alpine-magic-helpers#149] [Q&A]
I was wondering if it's possible to have access to the axios instance/variable used in `$fetch`. 
The idea is to have access to some features like [interceptors](https://github.com/axios/axios#interceptors) or [CancelToken](https://github.com/axios/axios#cancellation) from axios.

Basically something like this (I edited the js file to test):

![image](https://user-images.githubusercontent.com/14864288/126562344-8f8112a6-a245-4e44-8ea9-f71c6761ad6a.png)

**Comment by KevinBatdorf** [comment]
I think you can pass in the cancelToken as an option right? You would have to import it separately though

```js
import { CancelToken } from 'axios'
// ...
$fetch({
  // ...
  cancelToken: new CancelToken...
})

**Reply by KevinBatdorf** [reply]
If you have a reasonable PR though maybe we can merge it in. This repo is basically for Alpine V2 so not much is planned to change here.

**Reply by fgb4101** [reply]
Yeah, I was actually trying to avoid the duplicate import/load of the same dependency.
From what I saw in the dist js file, axios is there but private, I think is not tree-shaked, so it's complete.
About the PR, I will prepare one so you can take a look.

Aside question: it will be a new repo for Alpine v3?

**Reply by KevinBatdorf** [reply]
> Aside question: it will be a new repo for Alpine v3?

Maybe. Personally I'm still scoping the landscape and seeing what might be the best place to contribute. A few of the helpers aren't useful anymore, and a few others we can't easily recreate.
